server/test/check.go

```diff
@@ -1251,6 +1251,45 @@ var checkQueryTests = []checkQueryTest{
 			Allowed: true,
 		},
 	},
+	{
+		name:             "Check with TupleToUserset involving no object or userset",
+		resolveNodeLimit: defaultResolveNodeLimit,
+		request: &openfgapb.CheckRequest{
+			TupleKey: tuple.NewTupleKey("document:doc1", "viewer", "user:jon"),
+		},
+		typeDefinitions: []*openfgapb.TypeDefinition{
+			{
+				Type: "document",
+				Relations: map[string]*openfgapb.Userset{
+					"parent": {
+						Userset: &openfgapb.Userset_This{},
+					},
+					"viewer": {
+						Userset: &openfgapb.Userset_TupleToUserset{
+							TupleToUserset: &openfgapb.TupleToUserset{
+								Tupleset:        &openfgapb.ObjectRelation{Relation: "parent"},
+								ComputedUserset: &openfgapb.ObjectRelation{Relation: "viewer"},
+							},
+						},
+					},
+				},
+			},
+			{
+				Type: "folder",
+				Relations: map[string]*openfgapb.Userset{
+					"viewer": {
+						Userset: &openfgapb.Userset_This{},
+					},
+				},
+			},
+		},
+		tuples: []*openfgapb.TupleKey{
+			tuple.NewTupleKey("document:doc1", "parent", "folder1"), // folder1 isn't an object or userset
```

Comment:
> folder1 isn't an object or userset

But it is still valid, a user is able to write:
- `(user=anne, relation=viewer, object=folder:1)`
- `(user=folder:1, relation=parent, object=document:1)`

And expect `check(user=anne, relation=viewer, object=document:1)` to return `allowed: true`

Reply:
Right, but this isn't that. This is fixing

* `(user=anne, relation=viewer, object=folder:1)
* `(user=blah, relation=parent, object=document:doc1)`
which today returns an error, because the `parent` lookup on `blah` can't resolve because it is not associated with any object type, and thus an authz model cannot be loaded for it.

Reply:
In that case:
* is valid as it means objects of all types

```
(user=anne, relation=viewer, object=folder:1)
(user=*, relation=parent, object=document:doc1)
```
should return true as `*` is valid.

Re ignoring the user id here, that's cool, but can you update the test, because what it is testing is a very specific case that may or may not be indicative.

Write:
- `(user=anne, relation=viewer, object=folder:1)`
- `(user=folder:1, relation=parent, object=document:1)`
`check(user=anne, relation=viewer, object=document:1) // allowed=true`

Write:
- `(user=anne, relation=viewer, object=folder:1)`
- `(user=*, relation=parent, object=document:1)`
`check(user=anne, relation=viewer, object=document:1) // allowed=true`

Write:
- `(user=anne, relation=viewer, object=folder:1)`
- `(user=folder:1, relation=parent, object=document:1)`
- `(user=folder1, relation=parent, object=document:1)`
`check(user=anne, relation=viewer, object=document:1) // allowed=true`

Write:
- `(user=anne, relation=viewer, object=folder:1)`
- `(user=folder1, relation=parent, object=document:1)`
`check(user=anne, relation=viewer, object=document:1) // allowed=false`

Reply:
For the last test case you enumerated I'm going to save that work for the PR that address #235 . I'll submit a PR to address that work next.

I've added tests for the other cases that don't already have tests for.